### PR TITLE
Ext Docs Part 3 - Controllers

### PR DESCRIPTION
**Pull request by VSEphpbb**
Update the controllers doc.

Includes info copied from the wiki:
https://wiki.phpbb.com/Controller

This PR is part of the extension documentation overhaul:
Part 0 - Refactor index links #83 
Part 1 - Basics and Events #82 
Part 2 - Key Concepts #84 
Part 4 - Modules and Migrations #86 
Part 5 - Permissions #87 
Part 6 - Authentication #88 
Part 7 - Advanced Lessons #89 
PR Misc. #90 

This is for all doc branches (3.1 & 3.2)

### CODE REVIEW

**Comment by paul999** [comment]
Looks good to me.

File: development/extensions/tutorial_controllers.rst [hunk 1]
```diff
@@ -119,43 +125,66 @@ should look as follows:
...
 
-.. warning::
-
-    The order of the constructor arguments has to match the order in the service
-    definition.
+.. caution::
 
-There are a lot of other services defined by phpBB. You can find them in the
-``config/*.yml`` files of phpBB.
+    Remember the order of arguments must match the order of parameters in the
```
Comment (marc1706) [inline development/extensions/tutorial_controllers.rst]: IMHO either `Remember, the order ...` or `Remember that the order ...`

File: development/extensions/tutorial_controllers.rst [hunk 2]
```diff
@@ -165,55 +194,94 @@ footer:
...
-    acme_demo_route:
-        path: /demo/{name}
-        defaults: { _controller: acme.demo.controller:handle }
+To solve this, each controller must define a *route* in a ``config/routing.yml``
+file of the extension. This file is responsible for associating a controller's
+access name (i.e. what is typed in the URL) with its service (i.e. what we
+covered in `Dependencies`_).
 
+Recall that our controller expects a URL parameter to be passed to it as
+the ``$name`` variable. Therefor, we want our URL to look like:
```
Comment (marc1706) [inline development/extensions/tutorial_controllers.rst]: Therefore*

File: development/extensions/tutorial_controllers.rst [hunk 3]
```diff
@@ -225,6 +293,50 @@ the template engine and controller helper as dependencies:
             );
         }
 
+Next we will add a new method to the event listener which creates our link
+and assigns it to our template variable, as follows:
```
Comment (marc1706) [inline development/extensions/tutorial_controllers.rst]: The `as follows` is not really needed in this case.

**Comment by VSEphpbb** [comment]
Updated

**Pull request merged**